clients/src/main/java/org/apache/kafka/common/record/KafkaLZ4BlockOutputStream.java

```diff
@@ -259,8 +259,8 @@ private void ensureNotFinished() {
     @Override
     public void close() throws IOException {
         if (!finished) {
-            writeEndMark();
             flush();
+            writeEndMark();
```

Comment:
Have a point. We should call flush in consideration of actually may be not flushed EndMark after writeEndMark().
I add call flush() after writeEndMark().